projects/systelab-components/src/lib/contextmenu/abstract-context-menu.component.ts

```diff
@@ -78,4 +105,67 @@ export abstract class AbstractContextMenuComponent<T> extends AbstractContextCom
 			}
 		}
 	}
+
+	protected hideSubmenus (untilLevel: number): void {
+		if (untilLevel < this.lastMenuLevel) {
+			for (let i = this.lastMenuLevel; i > untilLevel; i--) {
+				this.toggle(this.previousShownMenu[i - 1]);
+				this.previousShownMenu.pop();
+				this.lastMenuLevel = i - 1;
+			}
+		}
+	}
+
+	public getMyLevel(actionId: string): number {
+		const actions: string[] = actionId.split(this.levelSeparator);
+		return actions.length - 1;
+	}
+
+	public getOptionDetailsActionId(actionId: string): string {
+		return this.getOption(actionId).actionId;
+	}
+
+	public getOptionDetailsHasChildren(actionId: string): boolean {
+		return this.getOption(actionId).hasChildren();
+	}
+
+
+	public showSubmenu(event: any, actionId: string, selectedChild: ElementRef, elementId: string): void {
+		const optionAcitionId = this.getOptionDetailsActionId(actionId);
+		const optionHasChildren = this.getOptionDetailsHasChildren(actionId);
+		const optionLevel = this.getMyLevel(actionId);
+
+		if (optionHasChildren) {
+			event.stopPropagation();
+			event.preventDefault();
+
+			if (this.previousActionChild !== optionAcitionId ) {
+				this.previousActionChild = optionAcitionId;
+				this.hideSubmenus(optionLevel);
+				this.lastMenuLevel = optionLevel + 1;
+
+				this.previousShownMenu.push(optionAcitionId + elementId);
+
+				this.toggle(optionAcitionId + elementId);
+
+				this.previousMenuWidth[this.lastMenuLevel - 1] = selectedChild.nativeElement.offsetWidth;
+
+				let leftPosition: number;
```

Comment:
Why don't you assign the variable to the value in a single line?

Reply:
Initally the code was different. In the final version, this has not sense.
Well seen.
Thanks